gems/pending/openstack/amqp/openstack_ceilometer_event_monitor.rb

```diff
@@ -0,0 +1,84 @@
+require 'openstack/openstack_event_monitor'
+require 'openstack/amqp/openstack_amqp_event'
+require 'openstack/amqp/openstack_ceilometer_event_converter'
+
+class OpenstackCeilometerEventMonitor < OpenstackEventMonitor
+  def self.available?(options = {})
+    begin
+      options[:ems].connect(:service => "Metering")
+      return true
+    rescue => ex
+      $log.debug("Skipping Openstack Ceilometer events. Availability check failed with #{ex}.") if $log
+    end
+    false
+  end
+
+  def self.plugin_priority
+    1
+  end
+
+  def initialize(options = {})
+    @options = options
+    @ems = options[:ems]
+  end
+
+  def start
+    @since          = nil
+    @monitor_events = true
+  end
+
+  def stop
+    @monitor_events = false
+  end
+
+  def provider_connection
+    @provider_connection ||= @ems.connect(:service => "Metering")
+  end
+
+  def each_batch
+    while @monitor_events
+      begin
+        $log.info("Quering Openstack Ceilometer for events newer than #{latest_event_timestamp}...") if $log
+        query_options = [{
```

Comment:
If we could use the API and query only whitelisted, that might be more efficient, when we will have large amount of events. But if that is not possible or tough to deliver in this release, then what you say will work.

I think instance.*, ... written in the settings might work better than the mapping, but whatever you think will be the best and easier to use :-)